webapp/src/ts/services/migrations.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+
+import { DbService } from '@mm-services/db.service';
+import { TargetCheckpointerMigration } from '@mm-services/migrations/target-checkpointer.migration';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MigrationsService {
+  constructor(
+    private dbService:DbService,
+    private targetCheckpointerMigration:TargetCheckpointerMigration,
+  ) {
+  }
+
+  private runningMigrationsPromise;
+  private runningMigrations = false;
+  private migrations = [
+    this.targetCheckpointerMigration
+  ];
+
+  private async run() {
+    for (const migration of this.migrations) {
+      if (!await migration.hasRun(this.dbService) && await migration.run()) {
+        await migration.setHasRun(this.dbService);
+      }
+    }
+    this.runningMigrations = false;
+  }
+
+  async runMigrations () {
+    if (this.runningMigrations) {
+      return await this.runningMigrationsPromise;
+    }
+
+    this.runningMigrations = true;
+    this.runningMigrationsPromise = this.run();
```

Comment:
Let's combine these two... if `runningMigrationsPromise` is set, then `runningMigrations` is true, right?

Reply:
It felt a little weird that resolving a promise would delete the reference to that promise, this is why I went with two variables. 
I'll change it and see how we feel.